### PR TITLE
[WIP] Ntnx whole deepakm ntnx

### PR DESCRIPTION
**Pull request by deepakm-ntnx**
*Issue #, if available:*

*Description of changes:*
PR got merge before completing the fixes https://github.com/nutanix-cloud-native/aws-eks-anywhere/pull/1
Here are the remaining fixes.
TODO post the test results here once complete

*Testing (if applicable):*
WIP
<pre>
LOCAL_E2E_TESTS=TestNutanixKubernetes120SimpleFlow BUNDLE_MANIFEST_URL=./bundle-release.yaml make local-e2e
LOCAL_E2E_TESTS=TestNutanixKubernetes121SimpleFlow BUNDLE_MANIFEST_URL=./bundle-release.yaml make local-e2e
LOCAL_E2E_TESTS=TestNutanixKubernetes122SimpleFlow BUNDLE_MANIFEST_URL=./bundle-release.yaml make local-e2e
LOCAL_E2E_TESTS=TestNutanixKubernetes123SimpleFlow BUNDLE_MANIFEST_URL=./bundle-release.yaml make local-e2e

LOCAL_E2E_TESTS=TestNutanixKubernetes120To121UbuntuUpgrade BUNDLE_MANIFEST_URL=./bundle-release.yaml make local-e2e
LOCAL_E2E_TESTS=TestNutanixKubernetes121To122UbuntuUpgrade BUNDLE_MANIFEST_URL=./bundle-release.yaml make local-e2e
LOCAL_E2E_TESTS=TestNutanixKubernetes122To123UbuntuUpgrade BUNDLE_MANIFEST_URL=./bundle-release.yaml make local-e2e

</pre>

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by thunderboltsid** [comment]
Some of these commits are from the previous PR; please rebase the branch against ntnx-whole.

File: pkg/providers/nutanix/envvars.go [hunk 1]
```diff
@@ -8,6 +8,15 @@ import (
...
 
 func setupEnvVars(datacenterConfig *anywherev1.NutanixDatacenterConfig) error {
+	creds := getCredsFromEnv()
+	if err := os.Setenv(nutanixUsernameKey, creds.username); err != nil {
+		return fmt.Errorf("unable to set %s: %v", nutanixUsernameKey, err)
+	}
+
+	if err := os.Setenv(nutanixPasswordKey, creds.password); err != nil {
+		return fmt.Errorf("unable to set %s: %v", nutanixPasswordKey, err)
+	}
```
Comment (thunderboltsid) [inline pkg/providers/nutanix/envvars.go]: What purpose does this serve here? `getCredsFromEnv` fetches the credentials from the environment variables that you are trying to set here.

Comment (deepakm-ntnx) [reply pkg/providers/nutanix/envvars.go]: initially i thought its required to pass the variables to container where clusterctl init command is actually run. tried without and seems not needed at this time. removed for timebeing, but we need to revisit the SetENvVar function as its called from 3 places and need to understand what all things we need to set here

**Comment by thunderboltsid** [comment]
Please do a `git rebase` instead of a `git merge`.

**Pull request closed without merging**